### PR TITLE
Add option to use a list of reduction strategies

### LINKED ISSUE #3: Server should be able to pick the reduction from a list
Currently in the configuration file we can put only one type of reduction at the time (see [reduction configuration](https://github.com/kumojin/extreme-carpaccio/blob/main/server/configuration.json#L4)). We should be able to use an array of reductions.

# Definition of done

- Replace the "reduction" field by an array in the configuration file
- If the picked reduction is not known we fallback to "STANDARD" (current behaviour) to prevent crash from typo in the config file.
- Add tests 

## To be discussed

- Do we randomly pick a reduction or we can weight them?

### PR DESCRIPTION
**Pull request by annemano**
**Add option to use a list of reduction strategies, instead of just one at a time**

resolves #3 

This PR makes it possible for the facilitator to either use one reduction strategy (as a string) or an array of strategies, which will then send one randomly picked from the array provided.

I've added unit tests (using TDD) for the different scenarios, a new `getReductionStrategy()` private function in the dispatcher and more information in the README.

I haven't worked with TS much, so let me know if there's something I should do better!

### CODE REVIEW

File: server/README-FR.md [hunk 1]
```diff
@@ -39,33 +39,49 @@ Les participants vont prendre du temps pour mettre en place leur machine, et cer
...
 
 Ensuite le facilitateur démarre le serveur et fait en sorte que toutes les équipes soient capables d'échanger des messages HTTP avec le serveur central. Une fois que tout le monde est prêt, le facilitateur autorise les équipes à commencer à implémenter leurs solutions (peut demander un redémarrage du serveur pour remettre les scores à 0) et les gens commencent à jouer.
 
+### Contraintes
+
 Durant la session, le facilitateur peut activer des "contraintes" depuis le fichier [configuration.json file](https://github.com/dlresende/extreme-carpaccio/blob/master/server/configuration.json), de façon à pimenter un peu le jeu et rajouter un peu de chaos. Quelques exemples : envoyer de mauvaises requêtes (**les participants doivent alors renvoyer une 400 - bad request**), changer la stratégie de réduction, changer le calcul des taxes, pénaliser le downtime etc. Tout changement au fichier de configuration est automatiquement pris en compte, sans nécessité de redémarrage du serveur. C'est à la discrétion du facilitateur d'annoncer ou pas les changements, dépendant de comment il souhaite mener la session.
 
+### Stratégies de réduction
```
Comment (annemano) [inline server/README-FR.md:71]: I think my linter has made several changes in the README, but this is the main change

File: server/apps/backend/src/services/dispatcher/Dispatcher.ts [hunk 2]
```diff
@@ -125,6 +125,16 @@ class Dispatcher {
     };
   }
 
+  private getReductionStrategy(): string | undefined {
+    const reductionStrategy = this.getConfiguration(this).reduction;
+    if (Array.isArray(reductionStrategy)) {
+      return reductionStrategy[
+        Math.floor(Math.random() * reductionStrategy.length)
```
Comment (xballoy) [inline server/apps/backend/src/services/dispatcher/Dispatcher.ts:132]: We are using [_.sample](https://lodash.com/docs/4.17.15#sample) at different places in the code to pick a random value from an array. You could use that instead.

File: server/apps/backend/src/services/dispatcher/Dispatcher.spec.ts [hunk 3]
```diff
@@ -35,23 +35,88 @@ describe('Dispatcher', () => {
...
-    jest.spyOn(dispatcher, 'sendOrderToSellers').mockImplementation(jest.fn());
+    it('should randomly send one of the reduction strategies when using an array', () => {
+      jest.spyOn(configuration, 'all').mockReturnValue({
+        reduction: ['HALF PRICE', 'PAY THE PRICE'],
+        badRequest: {
+          active: false,
+        },
+        active: true,
+      });
+      jest.spyOn(global.Math, 'random').mockReturnValue(0.5);
```
Comment (xballoy) [inline server/apps/backend/src/services/dispatcher/Dispatcher.spec.ts:67]: You are testing the implementation here, if you use `_.sample` (cf my comment on `getReductionStrategy`) your test won't work anymore.

There's no easy way to test randomness, maybe you could just test that you are getting a value from the array.

**Review by xballoy — APPROVED** [review]

File: server/apps/backend/src/services/dispatcher/Dispatcher.ts (hunk 2 above)
Comment (annemano) [reply server/apps/backend/src/services/dispatcher/Dispatcher.ts]: Good catch!

File: server/apps/backend/src/services/dispatcher/Dispatcher.spec.ts (hunk 3 above)
Comment (annemano) [reply server/apps/backend/src/services/dispatcher/Dispatcher.spec.ts]: You're right. I am now just testing that it correctly returns an element from the array.

**Pull request merged**